lldb/tools/lldb-server/lldb-platform.cpp

```diff
@@ -254,6 +445,65 @@ int main_platform(int argc, char *argv[]) {
   lldb_private::Args inferior_arguments;
   inferior_arguments.SetArguments(argc, const_cast<const char **>(argv));
 
+  if (fd != kInvalidFD) {
+    // Child process will handle the connection and exit.
+    Log *log = GetLog(LLDBLog::Platform);
+    if (!listen_host_port.empty()) {
+      LLDB_LOGF(log, "lldb-platform child: "
+                     "ambiguous parameters --listen and --fd");
+      return socket_error;
+    }
+
+    NativeSocket socket;
+#ifdef _WIN32
```

Comment:
And I think this could be a static method on the class I described above. `CompleteReceiving` ?